src/uu/tr/src/operation.rs

```diff
@@ -0,0 +1,569 @@
+
+    fn parse_char_range_backslash_collapse(input: &str) -> IResult<&str, Sequence> {
+        separated_pair(
+            preceded(tag("\\"), anychar),
+            tag("-"),
+            preceded(tag("\\"), anychar),
+        )(input)
+        .map(|(l, (a, b))| {
+            (l, {
+                let (start, end) = (u32::from(a), u32::from(b));
+                Sequence::CharRange(Box::new((start..=end).filter_map(std::char::from_u32)))
+            })
+        })
+    }
+
+    fn parse_char_range_octal_left(input: &str) -> IResult<&str, Sequence> {
+        separated_pair(
+            preceded(tag("\\"), recognize(many_m_n(1, 3, one_of("01234567")))),
+            tag("-"),
+            anychar,
+        )(input)
+        .map(|(l, (a, b))| {
+            (l, {
+                let (start, end) = (u32::from_str_radix(a, 8).unwrap(), u32::from(b));
+                Sequence::CharRange(Box::new((start..=end).filter_map(std::char::from_u32)))
+            })
+        })
+    }
+
+    fn parse_char_range_octal_right(input: &str) -> IResult<&str, Sequence> {
+        separated_pair(
+            anychar,
+            tag("-"),
+            preceded(tag("\\"), recognize(many_m_n(1, 3, one_of("01234567")))),
+        )(input)
+        .map(|(l, (a, b))| {
+            (l, {
+                let (start, end) = (u32::from(a), u32::from_str_radix(b, 8).unwrap());
+                Sequence::CharRange(Box::new((start..=end).filter_map(std::char::from_u32)))
+            })
+        })
+    }
+
+    fn parse_char_range_octal_leftright(input: &str) -> IResult<&str, Sequence> {
+        separated_pair(
+            preceded(tag("\\"), recognize(many_m_n(1, 3, one_of("01234567")))),
+            tag("-"),
+            preceded(tag("\\"), recognize(many_m_n(1, 3, one_of("01234567")))),
+        )(input)
+        .map(|(l, (a, b))| {
+            (l, {
+                let (start, end) = (
+                    u32::from_str_radix(a, 8).unwrap(),
+                    u32::from_str_radix(b, 8).unwrap(),
+                );
+                Sequence::CharRange(Box::new((start..=end).filter_map(std::char::from_u32)))
+            })
+        })
+    }
+
+    fn parse_char_star(input: &str) -> IResult<&str, Sequence> {
+        delimited(tag("["), anychar, tag("*]"))(input).map(|(l, c)| (l, Sequence::CharStar(c)))
+    }
+
+    fn parse_char_repeat(input: &str) -> IResult<&str, Sequence> {
+        delimited(
+            tag("["),
+            separated_pair(anychar, tag("*"), digit1),
+            tag("]"),
+        )(input)
+        .map(|(l, (c, n))| {
+            (
+                l,
+                Sequence::CharRange(Box::new(std::iter::repeat(c).take(n.parse().unwrap()))),
+            )
+        })
+    }
+
+    fn parse_alnum(input: &str) -> IResult<&str, Sequence> {
+        tag("[:alnum:]")(input).map(|(l, _)| {
+            (
+                l,
+                Sequence::CharRange(Box::new(('0'..='9').chain('A'..='Z').chain('a'..='z'))),
```

Comment:
Don't worry about this error. It will go away once https://github.com/uutils/coreutils/pull/2438 is merged which will increase the minimum supported rust version to 1.51.